tests/authenticate.bats

```diff
@@ -19,7 +19,7 @@ load helpers
   run_buildah 0 login --username testuserfoo --password testpassword docker.io
 
   run_buildah 125 logout --authfile /tmp/nonexistent docker.io
-  expect_output "checking authfile: stat /tmp/nonexistent: no such file or directory"
+  expect_output --substring "checking authfile: stat /tmp/nonexistent: no such file or directory"
```

Comment:
I see a lot of these `--substring` additions. These are bugs. These should not happen.

The root cause seems similar to the `buildah unshare` bug I reported earlier: some commands emit an unhelpful, unwanted extra "exit status" message:
```
$ bin/buildah logout --authfile /foo quay.io
checking authfile: stat /foo: no such file or directory
time="2022-03-07T09:54:01-07:00" level=error msg="exit status 125"
```
Can we fix the problem at the root, instead of working around it? @nalind do you know why this bug exists, and how to fix it?

Reply:
@edsantiago Sure. I'll check this issue seems to be with `buildah`. It logs `error` on `stderr` in case of `rootless` but does not in case of `root`. Not sure sure which one is the correct here but I'll make sure `rootless` buildah behaves same as `root`.

Reply:
@nalind Is it safe to remove error from `c/storage`, I have a feeling that something is using this `error` on `podman` side but just a feeling, I'll check this.

Reply:
Fixed here: https://github.com/containers/storage/pull/1162